proposals/base-relative-assets.md

```diff
@@ -0,0 +1,132 @@
+- Start Date: 2022-11-09
+- Reference Issues: <!-- related issues, otherwise leave empty -->
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+With this proposal it will be possible to build Astro sites that can be deployed to any path on any domain regardless of the base. This will be done through an new config option `build.linkStyle`.
+
+With this option you can tell Astro to build links (for assets like scripts an styles) to be absolute (the default and current behavior) or relative (relative to the base).
+
+# Example
+
+By default Astro will continue to add links with absolute paths from the `base`. For example with this config:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  base: '/docs',
+  build: {
+    linkStyle: 'absolute'
+  }
+});
+```
+
+Your CSS will build to this:
+
+```html
+<link rel="stylesheet" href="/docs/assets/main.hash.css">
+```
+
+Using the `relative` config option:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  build: {
+    linkStyle: 'relative'
+  }
+});
+```
+
+You'll instead get this:
+
+```html
+<link rel="stylesheet" href="../assets/main.hash.css">
+```
+
+# Motivation
+
+We often get support questions on how to deploy an Astro site without knowing the base in advance. There has been discussions and RFCs to solve this problem such as [this one](https://github.com/withastro/rfcs/discussions/334) and [this one](https://github.com/withastro/rfcs/discussions/23).
+
+People have tried to use `./` in the `base` option to make this happen to, such as in [this issue](https://github.com/withastro/astro/issues/4229).
```

Comment:
We might want a warning for when people do use `./` in `base` telling them to use `build.linkStyle` instead?

Reply:
I'm probably here to chime in that I prefer `base: './'` instead and not have a new option. The functionality overlaps so collapsing it might be easier to understand for the end-user. We can split this internally if it simplifies the internal code.

Reply:
Do you prefer `./` here because that's what Vite does? Does this work in a multi-page app for them? I'd love to see what this looks like.

Reply:
Yeah I partly prefer `./` because of Vite, and the idea of "the least config option" 😬 But I could be persuaded to `linkStyle` if it supports something that `./` can't.

`base: './'` does work in an MPA today but only for the client-side ([stackblitz](https://stackblitz.com/edit/vitejs-vite-nym34j?terminal=dev)).